commit f76e68c5cb679befb43ba71df7cc39f3840ca161
Author: Ryan Aslett <github@mixologic.com>
Date:   Wed Jan 14 06:55:59 2015 -0800

    fixed the file field defaults issue

diff --git a/multifield.module b/multifield.module
index 738c8e3..3ec4bf0 100644
--- a/multifield.module
+++ b/multifield.module
@@ -443,13 +443,20 @@ function multifield_item_serialize(&$item, $delta, $machine_name) {
       // @see field_sql_storage_field_storage_write()
       // @todo Should this be using array_key_exists() instead of isset()?
       if (!isset($item[$subfield_name][LANGUAGE_NONE][0][$column])) {
-        $item[$subfield_name][LANGUAGE_NONE][0][$column] = isset($details['default']) ? $details['default'] : NULL;
+        if (isset($details['default'])) {
+          $item[$subfield_name][LANGUAGE_NONE][0][$column] = $details['default'];
+        }
       }
 
       // We need to assign this value by reference because
       // $items[$delta][$subfield_name] could be modified in
       // multifield_field_insert() or multifield_field_update().
-      $item[$subfield_name . '_' . $column] = &$item[$subfield_name][LANGUAGE_NONE][0][$column];
+      if (isset($item[$subfield_name][LANGUAGE_NONE][0][$column])) {
+        $item[$subfield_name . '_' . $column] = &$item[$subfield_name][LANGUAGE_NONE][0][$column];
+      }
+      else {
+        $item[$subfield_name . '_' . $column] = NULL;
+      }
     }
     //unset($item[$subfield_name]);
   }
