diff --git a/multifield.module b/multifield.module
index 13a1d41..fc43f0d 100644
--- a/multifield.module
+++ b/multifield.module
@@ -405,7 +405,7 @@ function multifield_form_field_ui_field_overview_form_alter(&$form, &$form_state
 
   // Prevent multifields from being added to multifields themselves.
   $form['fields']['_add_new_field']['type']['#options'] = array_diff_key($form['fields']['_add_new_field']['type']['#options'], multifield_load_all());
-  if (isset($form['fields']['_add_existing_field'])){
+  if (isset($form['fields']['_add_existing_field'])) {
     $form['fields']['_add_existing_field']['field_name']['#options'] = array_diff_key($form['fields']['_add_existing_field']['field_name']['#options'], multifield_get_fields());
   }
 
@@ -516,3 +516,34 @@ function multifield_admin_menu_map() {
 
   return $map;
 }
+
+/**
+ * Implements hook_theme().
+ */
+function multifield_theme() {
+  return array(
+    'multifield' => array(
+      'render element' => 'element',
+    ),
+  );
+}
+
+/**
+ * Implements hook_preprocess_field().
+ *
+ * Adds a theme suggestions for multifield so we can theme multifields
+ * independently of their subordinate fields. *
+ */
+function multifield_preprocess_field(&$variables, $hook) {
+  $element = $variables['element'];
+
+  if ($variables['element']['#formatter'] == 'multifield_default') {
+    $variables['theme_hook_suggestions'] = array_merge($variables['theme_hook_suggestions'], array(
+    'multifield',
+    'multifield__' . $element['#field_type'],
+    'multifield__' . $element['#field_name'],
+    'multifield__' . $element['#bundle'],
+    'multifield__' . $element['#field_name'] . '__' . $element['#bundle'],
+    ));
+  }
+}
