diff --git a/public/sites/all/modules/contrib/multifield/multifield.module b/public/sites/all/modules/contrib/multifield/multifield.module
index a43edb8..b50889c 100644
--- a/public/sites/all/modules/contrib/multifield/multifield.module
+++ b/public/sites/all/modules/contrib/multifield/multifield.module
@@ -524,3 +524,34 @@ function multifield_admin_menu_map() {
 
   return $map;
 }
+
+/**
+ * Implements hook_theme().
+ */
+function multifield_theme() {
+  return array(
+    'multifield' => array(
+      'render element' => 'element',
+    ),
+  );
+}
+
+/**
+ * Implements hook_preprocess_field().
+ *
+ * Adds a theme suggestions for multifield so we can theme multifields
+ * independently of their subordinate fields. *
+ */
+function multifield_preprocess_field(&$variables, $hook) {
+  $element = $variables['element'];
+
+  if ($variables['element']['#formatter'] == 'multifield_default') {
+    $variables['theme_hook_suggestions'] = array_merge($variables['theme_hook_suggestions'], array(
+    'multifield',
+    'multifield__' . $element['#field_type'],
+    'multifield__' . $element['#field_name'],
+    'multifield__' . $element['#bundle'],
+    'multifield__' . $element['#field_name'] . '__' . $element['#bundle'],
+    ));
+  }
+}
